### ISSUE TITLE
Nullable Self-Assignment should warn

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#33228]
```C#
public static class C
{
#nullable enable
    static string test = test; // We should warn here, as `test` is null
}
```
